plugins/modules/site_workflow_manager.py

```diff
@@ -1301,81 +1433,71 @@ def get_diff_merged(self, config):
+                        while True:
+                            execution_details = self.get_execution_details(execution_id)
+                            if execution_details.get("status") == "SUCCESS":
+                                self.result['changed'] = True
+                                site_updated = True
+                                self.updated_site_list.append(site_name_hierarchy)
+                                self.log("Site - {0} Updated Successfully".format(site_name_hierarchy), "INFO")
+                                break
+                            elif execution_details.get("bapiError"):
+                                self.module.fail_json(msg=execution_details.get(
+                                    "bapiError"), response=execution_details)
+
                 except Exception as e:
                     self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")
-                    error_message = "Site - {0} does not need any update".format(site_name)
+                    error_message = "Site - {0} does not need any update".format(site_name_hierarchy)
                     return {"error_message": error_message}
 
-            if not self.site_requires_update():
-                self.update_not_needed_sites.append(site_name)
-                self.log("Site - {0} does not need any update".format(site_name), "INFO")
-
-                return self
-
-            if site_updated and response and isinstance(response, dict):
-                execution_id = response.get("executionId")
-                while True:
-                    execution_details = self.get_execution_details(execution_id)
-                    if execution_details.get("status") == "SUCCESS":
-                        self.result['changed'] = True
-                        break
-                    elif execution_details.get("bapiError"):
-                        self.module.fail_json(msg=execution_details.get(
-                            "bapiError"), response=execution_details)
-                        break
-
-                if site_updated:
-                    self.updated_site_list.append(site_name)
-                    self.log("Site - {0} Updated Successfully".format(site_name), "INFO")
-
-                else:
-                    site_exists = self.site_exists()
-                    if site_exists:
-                        self.created_site_list.append(site_name)
-                        self.log("Site '{0}' created successfully".format(site_name), "INFO")
+                site_exists, current_site = self.site_exists()
+                if site_exists:
+                    self.update_not_needed_sites.append(site_name_hierarchy)
+                    self.log("Site '{0}' created successfully".format(site_name_hierarchy), "INFO")
 
-            elif self.compare_dnac_versions(self.get_ccc_version(), "2.3.7.6") >= 0 and self.site_requires_update():
+            elif self.compare_dnac_versions(self.get_ccc_version(), "2.3.7.6") >= 0:
```

Comment:
else is not enough? do we need to compare again?

Reply:
Using  this`self.site_requires_update()` within that block, No need to comparison again